project-base/tests/FrontendApiBundle/Functional/Product/ProductsTest.php

```diff
@@ -121,4 +121,43 @@ private function getExpectedDataForNineteenthProduct(): array
 
         return $arrayExpected;
     }
+
+    public function testLastProduct(): void
+    {
+        $query = '
+            query {
+                products (last: 1) {
+                    edges {
+                        node {
+                            name
+                        }
+                    }
+                }
+            }
+        ';
+
+        $jsonExpected = $this->getExpectedDataForLastProduct();
+
+        $this->assertQueryWithExpectedJson($query, $jsonExpected);
```

Comment:
Maybe we should discuss the way we are writing tests for the API.

I like this one as it is easy to write but I am worried about bulletproof in future. Against this one I wrote more prozaic test which takes a bit more time to write/read but it is testing more than just product. https://github.com/shopsys/shopsys/pull/1960/commits/7b57ab628568b1e138308aeb88f9529862d32daf

Reply:
I wanted to use that, but that has been done in 9.1 and is not available in 9.0

Reply:
Ok, that is fair enough 👍